pkg/tuple/tuple_test.go

```diff
@@ -297,6 +299,39 @@ var testCases = []struct {
 		tupleFormat:    nil,
 		relFormat:      nil,
 	},
+	{
+		input:          "testns:" + superLongID + "#testrel@user:testusr",
+		expectedOutput: "testns:" + superLongID + "#testrel@user:testusr",
+		tupleFormat: makeTuple(
+			ObjectAndRelation("testns", superLongID, "testrel"),
+			ObjectAndRelation("user", "testusr", "..."),
+		),
+		relFormat: rel("testns", superLongID, "testrel", "user", "testusr", ""),
+	},
+	{
+		input:          "testns:foo#testrel@user:" + superLongID,
+		expectedOutput: "testns:foo#testrel@user:" + superLongID,
+		tupleFormat: makeTuple(
+			ObjectAndRelation("testns", "foo", "testrel"),
+			ObjectAndRelation("user", superLongID, "..."),
+		),
+		relFormat: rel("testns", "foo", "testrel", "user", superLongID, ""),
+	},
```

Comment:
let's add a test with superlong subject and resource object ids

Reply:
you mean together or...?

Reply:
yeah, a test that has both a long subject id and long resource id